src/node.cc

```diff
@@ -842,7 +842,15 @@ static std::atomic_bool init_called{false};
 
 int InitializeNodeWithArgs(std::vector<std::string>* argv,
                            std::vector<std::string>* exec_argv,
-                           std::vector<std::string>* errors) {
+                           std::vector<std::string>* errors,
+                           ProcessFlags::Flags flags) {
+  // Set some flags if only kDefaultFlags was passed. This can make API version
+  // transitions easier for embedders.
+  if (flags & EnvironmentFlags::kDefaultFlags) {
+    flags = static_cast<ProcessFlags::Flags>(
+        flags | ProcessFlags::kDisableStdioInheritance);
```

Comment:
Perhaps it would be better to just define the default flags in a separate global variable and just `|` that here instead? That way it would be easier to expand in the future.

However, I almost feel like `flags` should either just be completely overridden if the defaults are wanted as the default flags would have no way of specifically unsetting certain flags *or* maybe node could just export the default flags and people can use that as a starting point instead? An example of this would be node core setting a default flag but the embedder wants to use the defaults except for one particular flag. If node exported the defautls, embedders could easily just do something like `DEFAULT_FLAGS & ~ProcessFlags::kFooBarBaz`.

Reply:
This code follows the pattern used by `EnvironmentFlags`, but since embedders only care about disabling default settings it does feel unnecessary to have a `kDefaultFlags`. And it is actually confusing that `kDefaultFlags` is not a bitwise union of flags.

I'm changing the code to treat no flags as default behavior, which should be least surprising.